pkg/hooks/hooks_test.go

```diff
@@ -77,3 +79,62 @@ func testManifest() pods.Manifest {
 	builder.SetID(podId)
 	return builder.GetManifest()
 }
+
+func TestHookRunWithTimeout(t *testing.T) {
+	type sleepTest struct {
+		sleep      int
+		timeout    int
+		shouldPass bool
+	}
+
+	var sleepTests = []sleepTest{
+		{1, 2, true},
+		{2, 1, false},
```

Comment:
It doesn't matter if `sleep` takes seconds or not, if you can use a command like `sleep 10` and then time it out after `50ms` then that's fast. If you then make the other test just `echo hi`, it should also run quickly. It's okay if this test is only slow if the underlying code is broken, since presumably it won't be broken most of the time.